In [2]:
import pandas as pd
import os

In [3]:
total = []
for x in os.listdir("../data/주유소/"):
    total.append(pd.read_csv("../data/주유소/" + x , encoding='euc-kr'))
df_total = pd.concat(total)

In [4]:
df= pd.concat([pd.read_csv("../data/주유소/" + x , encoding='euc-kr') for x in os.listdir("../data/주유소/") ])

In [5]:
df.shape

(181714, 11)

In [6]:
df_total.shape

(181714, 11)

In [7]:
df.head(3)

번호      지역            상호              주소        기간     상표 셀프여부  \
0  A0006039  서울 강남구  (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190101  SK에너지   셀프   
1  A0006039  서울 강남구  (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190102  SK에너지   셀프   
2  A0006039  서울 강남구  (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190103  SK에너지   셀프   

   고급휘발유   휘발유    경유  실내등유  
0   1673  1465  1365     0  
1   1673  1465  1365     0  
2   1673  1465  1365     0

In [8]:
df1 = df[['지역', '상호', '상표']].drop_duplicates()

In [9]:
df1.상호.unique().size

507

In [10]:
df1.shape

(528, 3)

In [11]:
df1[df1['상호'] == '강산제2주유소']

지역       상호      상표
59432  서울 성북구  강산제2주유소   GS칼텍스
59293  서울 성북구  강산제2주유소  현대오일뱅크

In [12]:
df1[df1['상호'].duplicated()]

지역                   상호      상표
10343   서울 강북구             (주)석산에너지  현대오일뱅크
26373   서울 구로구    SK에너지(주)직영 개봉동주유소   SK에너지
28263   서울 구로구                신성주유소  현대오일뱅크
32943   서울 노원구                신성주유소   S-OIL
39828  서울 동대문구                신성주유소  현대오일뱅크
59975   서울 성북구                동일주유소   GS칼텍스
64316   서울 송파구                대성주유소   SK에너지
72291   서울 양천구                현대주유소   S-OIL
80680   서울 용산구   현대오일뱅크(주)직영 소월길주유소  현대오일뱅크
86540    서울 중구                세화주유소   알뜰주유소
89052   서울 중랑구            면목SELF주유소   SK에너지
89577   서울 중랑구               오천만주유소   S-OIL
4928    서울 강남구                역삼주유소   GS칼텍스
52459   서울 서초구              양재최고주유소   S-OIL
58396   서울 성북구       GS칼텍스(주) 종암주유소   GS칼텍스
59293   서울 성북구              강산제2주유소  현대오일뱅크
60629   서울 성북구                우리주유소   SK에너지
67908   서울 송파구    현대오일뱅크㈜직영 잠실셀프주유소  현대오일뱅크
69616   서울 양천구              가로공원주유소   SK에너지
71996   서울 양천구  현대오일뱅크(주)직영 양천셀프주유소  현대오일뱅크
82645   서울 은평구              번개셀프주유소    자가상표

In [13]:
df[df['상호'] == '강산제2주유소']

번호      지역       상호              주소        기간      상표 셀프여부  \
59432  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20190101   GS칼텍스   셀프   
59433  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20190102   GS칼텍스   셀프   
59434  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20190103   GS칼텍스   셀프   
59435  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20190104   GS칼텍스   셀프   
59436  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20190105   GS칼텍스   셀프   
...         ...     ...      ...             ...       ...     ...  ...   
59399  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20191227  현대오일뱅크   셀프   
59400  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20191228  현대오일뱅크   셀프   
59401  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20191229  현대오일뱅크   셀프   
59402  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20191230  현대오일뱅크   셀프   
59403  A0009492  서울 성북구  강산제2주유소  서울 성북구 보국문로 52  20191231  현대오일뱅크   셀프   

       고급휘발유   휘발유    경유  실내등유  
59432      0  1439  1338  1100  
59433      0  1439  1338  1100  
59434      0  1439  1338  1100  
59435      0  1418  1328  1100  
59436      0  1418  1328  1100  
...      ...   ...   ...   ...  
59399      0  1492  1337  1000  
59400      0  1492  1337  1000  
59401      0  1492  1337  1000  
59402      0  1492  1337  1000  
59403      0  1492  1337  1000  

[364 rows x 11 columns]

# 각 지역별 1위 브랜드 출력하기 

In [14]:
df_20191231 = df[df['기간'] == 20191231].copy()

In [15]:
df_20191231[['지역', '상호', '상표']].drop_duplicates()

지역                     상호     상표
183    서울 강남구           (주)동하힐탑셀프주유소  SK에너지
367    서울 강남구               (주)만정에너지  GS칼텍스
551    서울 강남구            (주)보성 세곡주유소  SK에너지
735    서울 강남구             (주)선진도곡주유소  S-OIL
919    서울 강남구  (주)소모에너지엔테크놀러지 쎈트럴주유소  GS칼텍스
...       ...                    ...    ...
90387  서울 중랑구                  우림주유소  GS칼텍스
90571  서울 중랑구           재원에너지㈜ 범아주유소  S-OIL
90755  서울 중랑구                    중랑교  SK에너지
90939  서울 중랑구          지에스칼텍스㈜ 대원주유소  GS칼텍스
91123  서울 중랑구          지에스칼텍스㈜ 소망주유소  GS칼텍스

[492 rows x 3 columns]

In [16]:
df_20191231_gu = df_20191231.groupby(['지역', '상표'], as_index=False)[['상호']].count()

In [17]:
df_20191231_gu.rename(columns={"상호" : "매장수"}, inplace=True)

In [18]:
검증 = df_20191231_gu.sort_values(by=['매장수'], ascending=False).groupby('지역').first()

In [19]:
검증

상표  매장수
지역                 
서울 강남구   SK에너지   16
서울 강동구   SK에너지    7
서울 강북구   SK에너지    5
서울 강서구   SK에너지   13
서울 관악구   SK에너지    5
서울 광진구   SK에너지    5
서울 구로구   S-OIL    6
서울 금천구   SK에너지    4
서울 노원구   GS칼텍스    6
서울 도봉구   S-OIL    6
서울 동대문구  SK에너지    8
서울 동작구   GS칼텍스    3
서울 마포구   SK에너지    6
서울 서대문구  SK에너지    7
서울 서초구   GS칼텍스   14
서울 성동구   SK에너지    7
서울 성북구   SK에너지    7
서울 송파구   SK에너지   13
서울 양천구   SK에너지   13
서울 영등포구  SK에너지   14
서울 용산구   SK에너지    9
서울 은평구   GS칼텍스    6
서울 종로구   SK에너지    4
서울 중구    GS칼텍스    4
서울 중랑구   GS칼텍스    5

In [20]:
df_20191231_gu.query(" 지역 == '서울 동작구' ")

지역      상표  매장수
49  서울 동작구   GS칼텍스    3
50  서울 동작구   S-OIL    1
51  서울 동작구   SK에너지    3
52  서울 동작구  현대오일뱅크    3

In [21]:
df_20191231.query(" 지역 == '서울 동작구' ")

번호      지역                   상호                      주소  \
41803  A0000411  서울 동작구       (주)성림에너지 남성주유소   서울 동작구 동작대로 135 (사당동)   
41987  A0000428  서울 동작구              21세기주유소  서울 동작구 시흥대로 616 (신대방동)   
42171  A0000434  서울 동작구       대성산업(주) 노량진주유소  서울 동작구 노량진로 172 (노량진동)   
42355  A0009553  서울 동작구                매일주유소    서울 동작구 상도로 139 (상도동)   
42539  A0000379  서울 동작구               살피재주유소          서울 동작구 상도로 334   
42723  A0000371  서울 동작구      지에스칼텍스(주) 대방주유소        서울 동작구 여의대방로 214   
42907  A0000368  서울 동작구                창원ENG     서울 동작구 사당로 59 (상도동)   
43091  A0000425  서울 동작구    현대오일뱅크(주)직영 동작주유소    서울 동작구 상도로 403 (상도동)   
43274  A0000426  서울 동작구  현대오일뱅크(주)직영 사당셀프주유소    서울 동작구 동작대로 73 (사당동)   
43458  A0000430  서울 동작구                  흑석동    서울 동작구 현충로 101 (흑석동)   

             기간      상표 셀프여부  고급휘발유   휘발유    경유  실내등유  
41803  20191231   GS칼텍스   셀프      0  1579  1429     0  
41987  20191231   SK에너지   셀프   1869  1599  1439     0  
42171  20191231   GS칼텍스   일반      0  1598  1458  1070  
42355  20191231   S-OIL   일반      0  1579  1479  1200  
42539  20191231   SK에너지   일반      0  1675  1545     0  
42723  20191231   GS칼텍스   일반   1739  1559  1399  1000  
42907  20191231  현대오일뱅크   일반      0  1564  1414     0  
43091  20191231  현대오일뱅크   일반   1779  1589  1479     0  
43274  20191231  현대오일뱅크   셀프   1744  1564  1414     0  
43458  20191231   SK에너지   셀프      0  1659  1515     0

# 12월 한달 평균 가장 휘발유 가격이 비싼 동네는?(지역구 기준)
## 과제

In [22]:
df_12 = df_total[(df_total['기간']).apply(lambda x : str(x)[:6]) == '201912']

In [33]:
df_12

번호      지역             상호              주소        기간     상표 셀프여부  \
153    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191201  SK에너지   셀프   
154    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191202  SK에너지   셀프   
155    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191203  SK에너지   셀프   
156    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191204  SK에너지   셀프   
157    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191205  SK에너지   셀프   
...         ...     ...            ...             ...       ...    ...  ...   
91119  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191227  GS칼텍스   셀프   
91120  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191228  GS칼텍스   셀프   
91121  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191229  GS칼텍스   셀프   
91122  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191230  GS칼텍스   셀프   
91123  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191231  GS칼텍스   셀프   

       고급휘발유   휘발유    경유  실내등유  
153     1792  1640  1495     0  
154     1792  1640  1495     0  
155     1792  1640  1495     0  
156     1792  1640  1495     0  
157     1792  1640  1495     0  
...      ...   ...   ...   ...  
91119      0  1540  1389  1100  
91120      0  1540  1389  1100  
91121      0  1540  1389  1100  
91122      0  1540  1389  1100  
91123      0  1540  1389  1100  

[15313 rows x 11 columns]

In [37]:
df_12.groupby('지역').mean().sort_values(by='휘발유', ascending=False)

기간        고급휘발유          휘발유           경유        실내등유
지역                                                                      
서울 중구    2.019122e+07  1940.164223  2065.598240  1885.480938  930.407625
서울 용산구   2.019122e+07  2214.119565  1951.595652  1816.254348  682.076087
서울 종로구   2.019122e+07  1686.376344  1899.917563  1737.727599  563.193548
서울 강남구   2.019122e+07  1689.507781  1773.900082  1619.884521  388.343980
서울 마포구   2.019122e+07   842.941748  1746.258900  1601.527508  555.608414
서울 성동구   2.019122e+07   802.151803  1649.977230  1480.445920  597.597723
서울 서초구   2.019122e+07  1343.990143  1643.736559  1499.260753  388.523297
서울 영등포구  2.019122e+07   533.050403  1641.550403  1484.408266  443.990927
서울 강동구   2.019122e+07   842.360887  1635.522177  1485.193548  741.497984
서울 서대문구  2.019122e+07   891.567742  1602.449462  1435.116129  356.666667
서울 노원구   2.019122e+07   593.638710  1601.903226  1446.266667  840.612903
서울 구로구   2.019122e+07   247.523810  1597.913978  1422.946237  659.116743
서울 관악구   2.019122e+07   458.372984  1591.328629  1441.721774  506.175403
서울 동작구   2.019122e+07   710.100000  1590.961290  1453.067742  325.903226
서울 양천구   2.019122e+07   628.036160  1571.936409  1407.501247  334.857855
서울 성북구   2.019122e+07   626.264045  1564.967697  1403.882022  607.946629
서울 송파구   2.019122e+07   794.521994  1564.572825  1415.964809  388.923754
서울 동대문구  2.019122e+07   242.099707  1563.275660  1385.655425  477.859238
서울 강서구   2.019122e+07   609.142586  1560.121673  1397.685361  377.151141
서울 금천구   2.019122e+07   130.342432  1559.215881  1393.905707  635.310174
서울 광진구   2.019122e+07   838.908918  1556.163188  1387.250474  399.882353
서울 은평구   2.019122e+07   609.607211  1547.113852  1391.865275  364.392789
서울 중랑구   2.019122e+07   230.320565  1538.842742  1371.245968  679.657258
서울 도봉구   2.019122e+07   406.367384  1532.709677  1381.449821  537.474910
서울 강북구   2.019122e+07   559.741935  1511.468983  1364.295285  222.580645

In [36]:
df_12.groupby('지역').mean().sort_values(by='휘발유', ascending=False)[['휘발유']]

휘발유
지역                  
서울 중구    2065.598240
서울 용산구   1951.595652
서울 종로구   1899.917563
서울 강남구   1773.900082
서울 마포구   1746.258900
서울 성동구   1649.977230
서울 서초구   1643.736559
서울 영등포구  1641.550403
서울 강동구   1635.522177
서울 서대문구  1602.449462
서울 노원구   1601.903226
서울 구로구   1597.913978
서울 관악구   1591.328629
서울 동작구   1590.961290
서울 양천구   1571.936409
서울 성북구   1564.967697
서울 송파구   1564.572825
서울 동대문구  1563.275660
서울 강서구   1560.121673
서울 금천구   1559.215881
서울 광진구   1556.163188
서울 은평구   1547.113852
서울 중랑구   1538.842742
서울 도봉구   1532.709677
서울 강북구   1511.468983

In [24]:
df_12.sort_values(by='휘발유', ascending=False)

번호      지역        상호                      주소        기간      상표  \
86697  A0000767   서울 중구     서남주유소            서울 중구 통일로 30  20191210   SK에너지   
86690  A0000767   서울 중구     서남주유소            서울 중구 통일로 30  20191203   SK에너지   
86691  A0000767   서울 중구     서남주유소            서울 중구 통일로 30  20191204   SK에너지   
86692  A0000767   서울 중구     서남주유소            서울 중구 통일로 30  20191205   SK에너지   
86693  A0000767   서울 중구     서남주유소            서울 중구 통일로 30  20191206   SK에너지   
...         ...     ...       ...                     ...       ...     ...   
50705  A0011826  서울 서초구  만남의광장주유소    서울 서초구 양재대로12길 73-71  20191225  알뜰(ex)   
80562  A0000212  서울 용산구     용일주유소  서울 용산구 한강대로 37 (한강로3가)  20191225   GS칼텍스   
80561  A0000212  서울 용산구     용일주유소  서울 용산구 한강대로 37 (한강로3가)  20191224   GS칼텍스   
60321  A0009496  서울 성북구    신동방주유소     서울 성북구 월계로 84 (장위동)  20191230   S-OIL   
80563  A0000212  서울 용산구     용일주유소  서울 용산구 한강대로 37 (한강로3가)  20191226   GS칼텍스   

      셀프여부  고급휘발유   휘발유    경유  실내등유  
86697   일반   2675  2332  2155  1610  
86690   일반   2675  2332  2155  1594  
86691   일반   2675  2332  2155  1594  
86692   일반   2675  2332  2155  1610  
86693   일반   2675  2332  2155  1610  
...    ...    ...   ...   ...   ...  
50705   셀프      0  1438  1320     0  
80562   일반      0     0  1538  1148  
80561   일반      0     0  1538  1148  
60321   일반      0     0  1358     0  
80563   일반      0     0  1538  1148  

[15313 rows x 11 columns]

In [25]:
df_12[df_12['지역'] == '서울 중구'].sort_values(by='휘발유', ascending=False)

번호     지역     상호              주소        기간     상표 셀프여부  고급휘발유  \
86693  A0000767  서울 중구  서남주유소    서울 중구 통일로 30  20191206  SK에너지   일반   2675   
86690  A0000767  서울 중구  서남주유소    서울 중구 통일로 30  20191203  SK에너지   일반   2675   
86692  A0000767  서울 중구  서남주유소    서울 중구 통일로 30  20191205  SK에너지   일반   2675   
86694  A0000767  서울 중구  서남주유소    서울 중구 통일로 30  20191207  SK에너지   일반   2675   
86695  A0000767  서울 중구  서남주유소    서울 중구 통일로 30  20191208  SK에너지   일반   2675   
...         ...    ...    ...             ...       ...    ...  ...    ...   
86898  A0009076  서울 중구  세화주유소  서울 중구 왕십리로 403  20191227  알뜰주유소   일반      0   
86899  A0009076  서울 중구  세화주유소  서울 중구 왕십리로 403  20191228  알뜰주유소   일반      0   
86900  A0009076  서울 중구  세화주유소  서울 중구 왕십리로 403  20191229  알뜰주유소   일반      0   
86901  A0009076  서울 중구  세화주유소  서울 중구 왕십리로 403  20191230  알뜰주유소   일반      0   
86902  A0009076  서울 중구  세화주유소  서울 중구 왕십리로 403  20191231  알뜰주유소   일반      0   

        휘발유    경유  실내등유  
86693  2332  2155  1610  
86690  2332  2155  1594  
86692  2332  2155  1610  
86694  2332  2155  1610  
86695  2332  2155  1610  
...     ...   ...   ...  
86898  1596  1398   995  
86899  1596  1398   995  
86900  1596  1398   995  
86901  1596  1398   995  
86902  1596  1398   995  

[341 rows x 11 columns]

### 강사님이 하심

In [31]:
df_201912 = df[df['기간'].apply(lambda x : str(x)[:6] == '201912')].copy()

In [32]:
df_201912

번호      지역             상호              주소        기간     상표 셀프여부  \
153    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191201  SK에너지   셀프   
154    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191202  SK에너지   셀프   
155    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191203  SK에너지   셀프   
156    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191204  SK에너지   셀프   
157    A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20191205  SK에너지   셀프   
...         ...     ...            ...             ...       ...    ...  ...   
91119  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191227  GS칼텍스   셀프   
91120  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191228  GS칼텍스   셀프   
91121  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191229  GS칼텍스   셀프   
91122  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191230  GS칼텍스   셀프   
91123  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191231  GS칼텍스   셀프   

       고급휘발유   휘발유    경유  실내등유  
153     1792  1640  1495     0  
154     1792  1640  1495     0  
155     1792  1640  1495     0  
156     1792  1640  1495     0  
157     1792  1640  1495     0  
...      ...   ...   ...   ...  
91119      0  1540  1389  1100  
91120      0  1540  1389  1100  
91121      0  1540  1389  1100  
91122      0  1540  1389  1100  
91123      0  1540  1389  1100  

[15313 rows x 11 columns]

In [40]:
df_201912_mean = df_201912.groupby("지역")[['휘발유']].mean().sort_values(by=['휘발유'], ascending=False)

In [41]:
df_201912_mean

휘발유
지역                  
서울 중구    2065.598240
서울 용산구   1951.595652
서울 종로구   1899.917563
서울 강남구   1773.900082
서울 마포구   1746.258900
서울 성동구   1649.977230
서울 서초구   1643.736559
서울 영등포구  1641.550403
서울 강동구   1635.522177
서울 서대문구  1602.449462
서울 노원구   1601.903226
서울 구로구   1597.913978
서울 관악구   1591.328629
서울 동작구   1590.961290
서울 양천구   1571.936409
서울 성북구   1564.967697
서울 송파구   1564.572825
서울 동대문구  1563.275660
서울 강서구   1560.121673
서울 금천구   1559.215881
서울 광진구   1556.163188
서울 은평구   1547.113852
서울 중랑구   1538.842742
서울 도봉구   1532.709677
서울 강북구   1511.468983

In [30]:
df_201912['휘발유'].mean()

1630.5358192385554

In [42]:
df_201912_mean.mean()

휘발유    1642.520079
dtype: float64

In [43]:
df_201912_mean['휘발유']

지역
서울 중구      2065.598240
서울 용산구     1951.595652
서울 종로구     1899.917563
서울 강남구     1773.900082
서울 마포구     1746.258900
서울 성동구     1649.977230
서울 서초구     1643.736559
서울 영등포구    1641.550403
서울 강동구     1635.522177
서울 서대문구    1602.449462
서울 노원구     1601.903226
서울 구로구     1597.913978
서울 관악구     1591.328629
서울 동작구     1590.961290
서울 양천구     1571.936409
서울 성북구     1564.967697
서울 송파구     1564.572825
서울 동대문구    1563.275660
서울 강서구     1560.121673
서울 금천구     1559.215881
서울 광진구     1556.163188
서울 은평구     1547.113852
서울 중랑구     1538.842742
서울 도봉구     1532.709677
서울 강북구     1511.468983
Name: 휘발유, dtype: float64

In [47]:
df_201912_mean[df_201912_mean['휘발유'] > df_201912_mean.mean().values[0]]

휘발유
지역                 
서울 중구   2065.598240
서울 용산구  1951.595652
서울 종로구  1899.917563
서울 강남구  1773.900082
서울 마포구  1746.258900
서울 성동구  1649.977230
서울 서초구  1643.736559

## 220428
### 서울시 소재 주유소 - 셀프 주유소, 일반 주유소 집계

In [51]:
df['셀프여부'].value_counts()

일반    108492
셀프     73222
Name: 셀프여부, dtype: int64

In [56]:
df['셀프여부'].unique()

array(['셀프', '일반'], dtype=object)

In [57]:
df

번호      지역             상호              주소        기간     상표 셀프여부  \
0      A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190101  SK에너지   셀프   
1      A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190102  SK에너지   셀프   
2      A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190103  SK에너지   셀프   
3      A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190104  SK에너지   셀프   
4      A0006039  서울 강남구   (주)동하힐탑셀프주유소  서울 강남구 논현로 640  20190105  SK에너지   셀프   
...         ...     ...            ...             ...       ...    ...  ...   
91119  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191227  GS칼텍스   셀프   
91120  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191228  GS칼텍스   셀프   
91121  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191229  GS칼텍스   셀프   
91122  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191230  GS칼텍스   셀프   
91123  A0032659  서울 중랑구  지에스칼텍스㈜ 소망주유소  서울 중랑구 망우로 475  20191231  GS칼텍스   셀프   

       고급휘발유   휘발유    경유  실내등유  
0       1673  1465  1365     0  
1       1673  1465  1365     0  
2       1673  1465  1365     0  
3       1673  1465  1365     0  
4       1673  1465  1365     0  
...      ...   ...   ...   ...  
91119      0  1540  1389  1100  
91120      0  1540  1389  1100  
91121      0  1540  1389  1100  
91122      0  1540  1389  1100  
91123      0  1540  1389  1100  

[181714 rows x 11 columns]

In [68]:
df_12['셀프여부'].value_counts()

일반    9011
셀프    6302
Name: 셀프여부, dtype: int64

In [84]:
df[['상호', '셀프여부']]

상호 셀프여부
0       (주)동하힐탑셀프주유소   셀프
1       (주)동하힐탑셀프주유소   셀프
2       (주)동하힐탑셀프주유소   셀프
3       (주)동하힐탑셀프주유소   셀프
4       (주)동하힐탑셀프주유소   셀프
...              ...  ...
91119  지에스칼텍스㈜ 소망주유소   셀프
91120  지에스칼텍스㈜ 소망주유소   셀프
91121  지에스칼텍스㈜ 소망주유소   셀프
91122  지에스칼텍스㈜ 소망주유소   셀프
91123  지에스칼텍스㈜ 소망주유소   셀프

[181714 rows x 2 columns]

In [83]:
df[['상호', '셀프여부']].drop_duplicates()['셀프여부'].value_counts()

일반    304
셀프    205
Name: 셀프여부, dtype: int64

In [85]:
df['상호'].unique().size

507

일반 + 셀프 = 509, 상호는 507 <br>
중간에 셀프로 바꾼곳이 있는 것 같다..

In [86]:
df[['상호', '셀프여부']].drop_duplicates().shape

(509, 2)

In [108]:
a = set(df[['상호', '셀프여부']].drop_duplicates()['상호'].tolist())

In [109]:
b = set(df['상호'].unique().tolist())

In [110]:
len(a)

507

In [111]:
len(b)

507

In [99]:
df_상셀듀 = df[['상호', '셀프여부']].drop_duplicates()

In [100]:
df_상셀듀[df_상셀듀['상호'].duplicated()]

상호 셀프여부
39828  신성주유소   일반
72291  현대주유소   셀프

In [127]:
df[['지역', '상호', '셀프여부']].drop_duplicates()['셀프여부'].value_counts()

일반    306
셀프    206
Name: 셀프여부, dtype: int64

In [128]:
df[['번호', '지역',  '상호', '셀프여부']].drop_duplicates()['셀프여부'].value_counts()

일반    306
셀프    206
Name: 셀프여부, dtype: int64

In [105]:
df[['번호', '지역', '상표', '상호', '셀프여부']].drop_duplicates()['셀프여부'].value_counts()

일반    314
셀프    214
Name: 셀프여부, dtype: int64

In [133]:
df_주셀 = df[['주소', '셀프여부']].drop_duplicates()['셀프여부'].value_counts()

In [123]:
df_주셀 / df_주셀.sum() * 100

일반    59.765625
셀프    40.234375
Name: 셀프여부, dtype: float64

In [130]:
df[['주소', '셀프여부']].drop_duplicates()

주소 셀프여부
0                서울 강남구 논현로 640   셀프
181       서울 강남구 봉은사로 433 (삼성동)   일반
362    서울특별시 강남구  헌릉로 731 (세곡동)   셀프
543        서울 강남구 도곡로 162 (도곡동)   일반
724        서울 강남구 삼성로 335 (대치동)   일반
...                         ...  ...
89866       서울 중랑구 동일로 881 (묵동)   일반
90047      서울 중랑구 망우로 229 (중화동)   셀프
90228            서울 중랑구 동일로 600   셀프
90409            서울 중랑구 망우로 475   셀프
66651           서울 송파구 위례서로 258   셀프

[512 rows x 2 columns]

In [132]:
df[['주소']].drop_duplicates()

주소
0                서울 강남구 논현로 640
181       서울 강남구 봉은사로 433 (삼성동)
362    서울특별시 강남구  헌릉로 731 (세곡동)
543        서울 강남구 도곡로 162 (도곡동)
724        서울 강남구 삼성로 335 (대치동)
...                         ...
89866       서울 중랑구 동일로 881 (묵동)
90047      서울 중랑구 망우로 229 (중화동)
90228            서울 중랑구 동일로 600
90409            서울 중랑구 망우로 475
66651           서울 송파구 위례서로 258

[512 rows x 1 columns]